content/news/2024-02-17-bevy-0.13/index.md

```diff
@@ -0,0 +1,3215 @@
+    Update,
+    (
+        system_with_commands,
+        another_system,
+    ).chain()
+);
+```
+
+This resolves a common beginner footgun: if two systems are ordered, shouldn't the second always see the results of the first?
+
+Automatically inserted `apply_deferred` systems are optimized by automatically merging them if
+possible. In most cases, it is recommended to remove all manually inserted
+`apply_deferred` systems, as allowing Bevy to insert and merge these systems as needed will
+usually be both faster and involve less boilerplate.
+
+```rust
+// This will only add one apply_deferred system.
+app.add_systems(
+    Update,
+    (
+        (system_1_with_commands, system_2).chain(),
+        (system_3_with_commands, system_4).chain(),
+    )
+);
+```
+
+If this new behavior does not work for you, please consult the migration guide.
+There are several new APIs that allow you to opt-out.
+
+## More Flexible One-Shot Systems
+
+<div class="release-feature-authors">authors: @Nathan-Fenner</div>
+
+In **Bevy 0.12**, we introduced [one-shot systems](https://bevyengine.org/news/bevy-0-12/#one-shot-systems), a handy way to call systems on demand without having to add them to a schedule.
+The initial implementation had some limitations with regards to what systems could and could not be used as one-shot systems.
+In **Bevy 0.13**, these limitations have been resolved.
+
+One-shot systems now support inputs and outputs.
+
+```rust
+fn increment_sys(In(increment_by): In<i32>, mut counter: ResMut<Counter>) -> i32 {
+    counter.0 += increment_by;
+    counter.0
+}
+
+let mut world = World::new();
+let id = world.register_system(increment_sys);
+
+world.insert_resource(Counter(1));
+let count_one = world.run_system_with_input(id, 5).unwrap(); // increment counter by 5 and return 6
+let count_two = world.run_system_with_input(id, 2).unwrap(); // increment counter by 2 and return 8
+```
+
+Running a system now returns the system output as `Ok(output)`. Note that output cannot be returned when calling one-shot systems through commands, because of their deferred nature.
+
+Exclusive systems can now be registered as one-shot systems:
+
+```rust
+world.register_system(|world: &mut World| { /* do anything */ });
+```
+
+Boxed systems can now be registered with `register_boxed_system`.
+
+These improvements round out one-shot systems significantly: they should now work just like any other Bevy system.
+
+## wgpu 0.19 Upgrade and Rendering Performance Improvements
+
+<div class="release-feature-authors">authors: @Elabajaba, @JMS55</div>
+
+In **Bevy 0.13** we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to [compile shaders asynchronously](https://github.com/bevyengine/bevy/pull/10812) to avoid shader compilation stutters and [multithread draw call creation](https://github.com/bevyengine/bevy/pull/9172 ) for better performance.
```

Comment:
```suggestion
In **Bevy 0.13** we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to [compile shaders asynchronously](https://github.com/bevyengine/bevy/pull/10812) to avoid shader compilation stutters and [multithread draw call creation](https://github.com/bevyengine/bevy/pull/9172 ) for better performance in CPU-bound scenes.
```